### DISCUSSION TITLE
PyRad/RadiPy vs Matlab engine for python

### DISCUSSION THREAD

**Discussion by chh105** [e0404/matRad#559] [General]
Hello,

I recently noticed the project to implement a python version of matRad. I think it would be really great to have a native python version for a number of reasons. However, I also just wanted to mention that python has a way to call matlab commands ([matlab engine for python](https://www.mathworks.com/help/matlab/matlab-engine-for-python.html)). If you're only interested in wrapping matRad function to be called in python, maybe the matlab engine thing could be a good option. I have also written code to do this for previous projects and am happy to share/link it if it's helpful at all

**Comment by wahln** [comment]
Yes, there's indeed a project going on in the background (but it has just started, there's barely any code yet). We also checked the matlab engine for python, as well as oct2py which works in a similar way with Octave. It is actually the idea to build a compatible package, which can, for example, call the dose calculation from matRad using the engine and then do optimization in python.
If you are interested to work on this / share your thoughts with us, let me know here shortly and I add you as a collaborator in the (currently still private) repository.
